### DISCUSSION TITLE
serialize and deserialize via fastcdr

### DISCUSSION THREAD

**Discussion by jfergusonmi** [eProsima/Fast-DDS#5400] [Support]
I am trying to convert fastdds DynamicData into a cdr buffer via fastcdr and convert a cdr buffer into DynamicData on the other side. Looking through the code and found some things like 

        eprosima::fastcdr::FastBuffer

and
        eprosima::fastcdr::Cdr

What would be an example of doing the convertions above mentioned for DynamicData?

**Comment by MiguelCompany** [comment]
This is exactly what [DynamicPubSubType](https://github.com/eProsima/Fast-DDS/blob/b4146214aa8eea772746e17ac486e75340908a3a/src/cpp/fastdds/xtypes/dynamic_types/DynamicPubSubType.cpp) does.

You could either use a `DynamicPubSubType` directly (the [header](https://github.com/eProsima/Fast-DDS/blob/b4146214aa8eea772746e17ac486e75340908a3a/include/fastdds/dds/xtypes/dynamic_types/DynamicPubSubType.hpp) is public), or you could take a look at the `serialize` and `deserialize` methods as an example if you want to include any customization of the code there.
